### ISSUE TITLE
Propagate changes to astropy affiliated package template

### ISSUE THREAD

**Issue by dkirkby** [dkirkby/speclite#18]
See changelog [here](https://github.com/astropy/package-template/blob/master/TEMPLATE_CHANGES.md).  This is needed for py3.5 compatibility (astropy/package-template#169) and for Javier's tutorial (fjaviersanchez/JupyterTutorial#10).

**Comment by dkirkby** [comment]
First check that I can reproduce the problem.  The first-time setup is:
```
conda create -n py35 python=3.5 anaconda
```
Next, in a new shell:
```
source activate py35
cd .../speclite
python setup.py install
```
The error is:
```
Traceback (most recent call last):
  File "setup.py", line 25, in <module>
    conf = config.ConfigParser()
AttributeError: module 'distutils.config' has no attribute 'ConfigParser'
```

**Comment by dkirkby** [comment]
I already have a remote pointing to the original template repo:
```
% git remote get-url template
https://github.com/astropy/package-template.git
```
Pull in the changes:
```
git fetch template
git merge template/master
```
There are a bunch of conflicts:
```
Auto-merging speclite/conftest.py
Auto-merging speclite/__init__.py
CONFLICT (content): Merge conflict in speclite/__init__.py
Auto-merging setup.py
CONFLICT (content): Merge conflict in setup.py
Auto-merging setup.cfg
CONFLICT (content): Merge conflict in setup.cfg
CONFLICT (modify/delete): TEMPLATE_CHANGES.md deleted in HEAD and modified in template/master. Version template/master of TEMPLATE_CHANGES.md left in tree.
Auto-merging README.rst
CONFLICT (content): Merge conflict in README.rst
Auto-merging .travis.yml
CONFLICT (content): Merge conflict in .travis.yml
Auto-merging .gitignore
Automatic merge failed; fix conflicts and then commit the result.
```

**Comment by dkirkby** [comment]
Use HEAD for: `__init__.py`, `setup.cfg`, `README.rst`

Edit to resolve conflicts for: `setup.py`, `.travis.yml`

Ignore files that (re)created by the merge: `TEMPLATE_CHANGES.md`, `packagename/*`

Stage the changes:
```
git add .travis.yml README.rst setup.cfg setup.py speclite/__init__.py
git add astropy_helpers
git rm -rf packagename
git rm TEMPLATE_CHANGES.md
```

Git status now reports:
```
Changes to be committed:

	modified:   .gitignore
	modified:   .travis.yml
	modified:   MANIFEST.in
	modified:   ah_bootstrap.py
	new file:   appveyor.yml
	modified:   docs/Makefile
	modified:   docs/conf.py
	modified:   ez_setup.py
	modified:   setup.py
	modified:   speclite/conftest.py
```
Finally, commit the changes:
```
git commit -m 'Update to latest astropy affiliated package template'
```

**Comment by dkirkby** [comment]
Update `CHANGES.rst` to summarize the changes:
- Update to latest astropy affiliate package template.
- Drop support for python 2.6 and add support for python 3.5.
- Add testing against LTS release of astropy.
- Drop testing against numpy 1.6 and add numpy 1.11.

**Comment by dkirkby** [comment]
I accidentally clobbered the scipy and pyyaml installs when I updated `.travis.yml` and picked up an un-necessary `Cython` dependency.  Update as follows (see [here](https://github.com/astropy/ci-helpers#travis) for details on the env vars used in `.travis.yml`):
```
- CONDA_DEPENDENCIES='scipy pyyaml matplotlib'
```
Should I set this var for the tests that use matplotlib?
```
$SETUP_XVFB: if True this makes sure e.g., interactive matplotlib backends work
  by starting up a X virtual framebuffer.
```
Push changes without this and see if the PR tests work...

**Comment by dkirkby** [comment]
Fixed in #19
